app/vmctl/vm_native.go

```diff
@@ -381,7 +381,9 @@ func buildMatchWithFilter(filter string, metricName string) (string, error) {
 			}
 			a = append(a, tf.String())
 		}
-		a = append(a, nameFilter)
+		if nameFilter != "" {
```

Comment:
It doesn't look to me like it solves the problem. Could you please add a test case to verify this?

Reply:
done

Reply:
I checked the new changes, and they work as the previous realization. Please check the screen below
<img width="1800" alt="Screenshot 2024-10-24 at 11 31 30" src="https://github.com/user-attachments/assets/b3857acb-6457-45f9-8852-7940133dcc03">

Reply:
updated logic and tests